tools/codegen/gen_backend_stubs.py

```diff
@@ -123,7 +123,7 @@ def error_on_missing_kernels(
         native_functions: Sequence[NativeFunction],
         backend_indices: Dict[DispatchKey, BackendIndex],
         backend_key: DispatchKey,
-        autograd_key: DispatchKey,
+        autograd_key: Optional[DispatchKey],
```

Comment:
I suppose these script changes are just refactoring and nothing specific to maxpool_2d?

Reply:
oh, initially I tried to codegen both 2d and 3d.  I abandoned 3d only bc i didn't want to write the shape function for now, 2d is commonly used, and has meta kernel.

When I codegenned both, i deleted the whole 'Autograd' section from ts_native_functions.yaml.  I discovered that doing this broke the codegen, since it did not handle an empty list here.  So these changes are to improve the codegen to handle a case where there are no Autograd ops for a backend.